pubsub/google/cloud/pubsub/topic.py

```diff
@@ -441,6 +468,21 @@ def publish(self, message, **attrs):
             {'data': message,
              'attributes': attrs})
 
+        # If too much time has elapsed since the first message
+        # was added, autocommit.
+        if self._max_interval < self.INFINITY:
+            now = time.time()
+            if now - self._start_timestamp > self._max_interval:
+                self.commit()
+                self._start_timestamp = now
+                return
+
+        # If the number of messages on the list is greater than the
+        # maximum allowed, autocommit (with the batch's client).
+        if len(self.messages) >= self._max_messages:
+            self.commit()
+            return
```

Comment:
This return unnecessary since it's the last line in the method. But maybe it's good to keep it to future proof?